internal/integration/roundtrip.go

```diff
@@ -126,3 +136,24 @@ func newRoundtripGetClient(t *testing.T, endpoint string) graphql.Client {
 		t:         t,
 	}
 }
+
+type MyDialer struct {
```

Comment:
Thinking out loud: would it be useful to test against multiple clients here? e.g. `x/net/websocket` or `nhooyr.io/websocket` or whatever is up there on grank.io. Not a big deal if there are issues but would help prove out if the interface can actually support different clients.

(If `x/net/websocket` works out, we could also use that in the example (with a comment suggesting others) and keep the main `go.mod` a bit smaller.)

Reply:
I tested with `x/net/websocket` and it works, I will add to the doc both ways: with `x/net/websocket` and with `gorilla/websocket`.
The package `gorilla/websocket` was in the `go.mod` because it was the client I used for the tests.
But now I can change and use `x/net/websocket` in the tests if you prefer, even if it adds `golang.org/x/net` as direct dependency, and `golang.org/x/sys` as indirect.